compiler/ml/typecore.ml

```diff
@@ -3409,119 +3404,9 @@ and type_label_exp ?type_clash_context create env loc ty_expected
 
 and type_argument ?type_clash_context ?recarg env sarg ty_expected' ty_expected
     =
-  (* ty_expected' may be generic *)
-  let no_labels ty =
-    let ls, tvar = list_labels env ty in
-    (not tvar) && List.for_all (fun x -> x = Nolabel) ls
-  in
-  let rec is_inferred sexp =
-    match sexp.pexp_desc with
-    | Pexp_ident _ | Pexp_apply _ | Pexp_field _ | Pexp_constraint _
-    | Pexp_coerce _ | Pexp_send _ | Pexp_new _ ->
-      true
-    | Pexp_sequence (_, e) | Pexp_open (_, _, e) -> is_inferred e
-    | Pexp_ifthenelse (_, e1, Some e2) -> is_inferred e1 && is_inferred e2
-    | _ -> false
-  in
-  match expand_head env ty_expected' with
-  | {desc = Tarrow (Nolabel, ty_arg, ty_res, _); level = _}
```

Comment:
This was not triggered since switching to uncurried by default, as a function is wrapped in `function$`.